trappy/ftrace.py

```diff
@@ -49,7 +49,7 @@ def _plot_freq_hists(allfreqs, what, axis, title):
 SPECIAL_FIELDS_RE = re.compile(
                         r"^\s*(?P<comm>.*)-(?P<pid>\d+)(?:\s+\(.*\))"\
                         r"?\s+\[(?P<cpu>\d+)\](?:\s+....)?\s+"\
-                        r"(?P<timestamp>[0-9]+\.[0-9]+): (\w+:\s+)+(?P<data>.+)"
+                        r"(?P<timestamp>[0-9]+(\.[0-9]+)?): (\w+:\s+)+(?P<data>.+)"
```

Comment:
We need to detect if the timestamps are in the ns format and then convert the timestamps to seconds, otherwise the data will look all wrong.